src/librustc/ty/constness.rs

```diff
@@ -83,15 +88,31 @@ impl<'tcx> TyCtxt<'tcx> {
         }
 
         if self.features().staged_api {
-            // in order for a libstd function to be considered min_const_fn
-            // it needs to be stable and have no `rustc_const_unstable` attribute
-            match self.lookup_stability(def_id) {
-                // stable functions with unstable const fn aren't `min_const_fn`
-                Some(&attr::Stability { const_stability: Some(_), .. }) => false,
-                // unstable functions don't need to conform
-                Some(&attr::Stability { ref level, .. }) if level.is_unstable() => false,
-                // everything else needs to conform, because it would be callable from
-                // other `min_const_fn` functions
+            // In order for a libstd function to be considered min_const_fn
+            // it needs to be stable and have no `rustc_const_unstable` attribute.
+            match self.lookup_const_stability(def_id) {
+                // `rustc_const_unstable` functions don't need to conform.
+                Some(&attr::ConstStability { ref level, .. }) if level.is_unstable() => false,
+                None => if let Some(stab) = self.lookup_stability(def_id) {
+                    if stab.level.is_stable() {
+                        self.sess.span_err(
+                            self.def_span(def_id),
+                            "stable const functions must have either `rustc_const_stable` or \
+                            `rustc_const_unstable` attribute",
+                        );
+                        // While we errored above, because we don't know if we need to conform, we
+                        // err on the "safe" side and require min_const_fn.
+                        true
+                    } else {
+                        // Unstable functions need not conform to min const fn.
+                        false
+                    }
+                } else {
+                    // Unstable functions or internal functions need not conform to min const fn.
+                    false
+                },
+                // Everything else needs to conform, because it would be callable from
+                // other `min_const_fn` functions.
                 _ => true,
```

Comment:
No, internal things need no annotation, if there's no annotation it must be min_const_fn

Reply:
```rust
                     // Unstable functions or internal functions need not conform to min const fn. 
```
You literally have a branch => `false` above. =P

Reply:
oh... I guess... Well we need to make a decision in either direction (default not min const fn or default min const fn). I don't think the attribute is misnamed in either variant

Reply:
I think we should err on the side of caution and make the default min_const_fn and then you can explicitly opt out with `#[rustc_const_unstable(...)]`.

Reply:
wait how is that more cautious? You can't call it from stable code anyway. With this PR there's no way you can accidentally expose its constness to users without an annotation

Reply:
Well maybe "cautious" isn't the right term. By making min_const_fn the default, it's more likely that more things will be ready for usage internally in min_const_fn stable stuff and it's less likely that we regress that potential (I say potential because it might not be utilized stably *yet*).

Reply:
I changed the default